apollo-router/tests/integration/query_planner.rs

```diff
@@ -185,21 +185,13 @@ async fn progressive_override_with_new_qp() {
         .build()
         .await;
     router.start().await;
-    router
-        .assert_log_contains(
-            "could not create router: \
-             failed to initialize the query planner: \
-             `experimental_query_planner_mode: new` or `both` cannot yet \
-             be used with progressive overrides. \
-             Remove uses of progressive overrides to try the experimental query planner, \
-             otherwise switch back to `legacy` or `both_best_effort`.",
-        )
-        .await;
-    router.assert_shutdown().await;
+    router.assert_started().await;
+    router.execute_default_query().await;
+    router.graceful_shutdown().await;
 }
 
 #[tokio::test(flavor = "multi_thread")]
-async fn progressive_override_with_legacy_qp_change_to_new_qp_keeps_old_config() {
+async fn progressive_override_with_legacy_qp_change_to_new_qp_updates_config() {
```

Comment:
do we still need those tests? previously we had them to ensure stuff works in JS and is not supported in RS. Since now it is supported -> should we drop those tests?